catalog/spatial/registry/registry-rest-endpoint/src/test/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpointTest.java

```diff
@@ -103,6 +106,128 @@ public void testReportWithRegistryPackage() throws ParserException, FederationAd
     }
 
     @Test
+    public void testFullReportWithIOException() throws IOException, FederationAdminException {
+
+        when(registryReportBuilder.getHTMLFromRegistryPackage(any(RegistryPackageType.class),
+                anyString())).thenThrow(new IOException());
+        when(federationAdminService.getRegistryObjectByRegistryId(anyString(),
+                anySet())).thenReturn(registryPackage);
+        Response response = restEndpoint.viewRegInfoHtml("metacardId", Arrays.asList("sourceIds"));
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()));
+    }
+
+    @Test
+    public void testSectionReportWithBlankMetacard() {
+        RegistryRestEndpoint reportViewer = new RegistryRestEndpoint();
+        Response response = reportViewer.viewSectionInfoHtml(null, null, null);
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.BAD_REQUEST.getStatusCode()));
+    }
+
+    @Test
+    public void testSectionReportWithFederationAdminException() throws FederationAdminException {
+        when(federationAdminService.getRegistryObjectByRegistryId(anyString(), anySet())).thenThrow(
+                new FederationAdminException());
+        Response response = restEndpoint.viewSectionInfoHtml("metacardId",
+                Arrays.asList("sourceIds"),
+                "section");
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()));
+    }
+
+    @Test
+    public void testSectionReportWithUnknownSectionName() throws FederationAdminException {
+        when(federationAdminService.getRegistryObjectByRegistryId(anyString(),
+                anySet())).thenReturn(registryPackage);
+        Response response = restEndpoint.viewSectionInfoHtml("metacardId",
+                Arrays.asList("sourceIds"),
+                "unknownSectionName");
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.BAD_REQUEST.getStatusCode()));
+    }
+
+    @Test
+    public void testSectionReportWithSectionName()
+            throws ParserException, FederationAdminException {
+        when(federationAdminService.getRegistryObjectByRegistryId(anyString(),
+                anySet())).thenReturn(registryPackage);
+
+        Response response = restEndpoint.viewSectionInfoHtml("metacardId",
+                Arrays.asList("sourceIds"),
+                "organizations");
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.OK.getStatusCode()));
+
+    }
+
+    @Test
+    public void testSectionReportWithIOException() throws IOException, FederationAdminException {
```

Comment:
After looking through the unit tests, the main reason the exceptions are being thrown are because some`RegistryRestEndpoint` methods call methods from mocked classes which contain these exceptions.  Some of the tests already were written and I used that as a foundation, let me know if you have any suggestions for alternative ways or if I should keep it as is.